egov-searcher/src/main/resources/billing-1.1-search.yml

```diff
@@ -1,26 +1,77 @@
+            b.isactive AS b_isactive, b.iscancelled AS b_iscancelled, 
+            b.createdby AS b_createdby, b.status as b_status, 
+            b.createddate AS b_createddate, b.lastmodifiedby AS b_lastmodifiedby, 
+            b.lastmodifieddate AS b_lastmodifieddate, bd.id AS bd_id, 
+            bd.billid AS bd_billid, bd.tenantid AS bd_tenantid,
+            bd.businessservice AS bd_businessservice, bd.demandid, 
+            bd.fromperiod, bd.toperiod, 
+            bd.billno AS bd_billno, bd.billdate AS bd_billdate, 
+            bd.consumercode AS bd_consumercode, bd.consumertype AS bd_consumertype, 
+            bd.billdescription AS bd_billdescription, bd.displaymessage AS bd_displaymessage, 
+            bd.minimumamount AS bd_minimumamount, bd.totalamount AS bd_totalamount, 
+            bd.callbackforapportioning AS bd_callbackforapportioning, bd.expirydate AS bd_expirydate, 
+            bd.partpaymentallowed AS bd_partpaymentallowed, bd.isadvanceallowed as bd_isadvanceallowed, 
+            bd.collectionmodesnotallowed AS bd_collectionmodesnotallowed, bd.createddate as bd_createddate, 
+            b.additionaldetails as b_additionaldetails,  bd.additionaldetails as bd_additionaldetails,
+            ad.id AS ad_id, ad.tenantid AS ad_tenantid, 
+            ad.billdetail AS ad_billdetail, ad.glcode AS ad_glcode, 
+            ad.orderno AS ad_orderno, ad.accountdescription AS ad_accountdescription, 
+            ad.amount AS ad_amount, ad.adjustedamount AS ad_adjustedamount, 
+            ad.taxheadcode AS ad_taxheadcode, ad.demanddetailid, 
+            ad.isactualdemand AS ad_isactualdemand, ad.purpose AS ad_purpose,   
+            ad.additionaldetails as ad_additionaldetails,
+            ptadd.doorNo as ptadd_doorNo, ptadd.addressline1 as ptadd_addressline1, 
+            ptadd.addressline2 as ptadd_addressline2, ptadd.landmark as ptadd_landmark, 
+            ptadd.city as ptadd_city, ptadd.pincode as ptadd_pincode, 
+            ptadd.locality as ptadd_locality, ptown.userid as ptown_userid 
+            
+            FROM egbs_bill_v1 b INNER JOIN egbs_billdetail_v1 bd ON b.id = bd.billid AND b.tenantid = bd.tenantid 
+            INNER egbs_billaccountdetail_v1 ad ON bd.id = ad.billdetail AND bd.tenantid = ad.tenantid 
+            LEFT OUTER JOIN eg_pt_property_v2 pt ON bd.consumercode = pt.propertyid 
+            INNER JOIN eg_pt_address_v2 ptadd ON ptadd.property = pt.propertyid INNER JOIN eg_pt_propertydetail_v2 ptdet ON ptdet.property = pt.propertyid 
+            INNER JOIN eg_pt_owner_v2 ptown ON ptdet.assessmentnumber = ptown.propertydetail $where AND b.status = 'ACTIVE')
 
+      SELECT *, tax.collectedamount as bd_collectedamount from bill_data bd INNER JOIN (select sum(ad_adjustedamount) as collectedamount, demandid from bill_data group by demandid) tax ON         tax.demandid=bd.demandid
+   searchParams: 
+    condition: AND
+    params:
+    - name: b.tenantid
+      isMandatory: true
```

Comment:
Allows for data dump including PII of the entire tenant. Bill also seems to be holding PII such as mobile address etc in its tables instead of having it in user service?

Reply:
Bills can be viewed by the public if they have any of the search parameters. mobile number and user name are commonly associated with bills in any billing system. if I have the account Id then payment page will show the persons he owner info. IF there are concerns please take it up with the leadership.